mypyc/irbuild/classdef.py

```diff
@@ -498,6 +498,9 @@ def populate_non_ext_bases(builder: IRBuilder, cdef: ClassDef) -> Value:
                 if builder.options.capi_version < (3, 8):
                     # TypedDict was added to typing in Python 3.8.
                     module = "typing_extensions"
+                    # It needs to be "_TypedDict" on typing_extensions 4.7.0+
+                    # and "TypedDict" otherwise.
```

Comment:
If this breaks tests with typing_extensions <4.7 installed, should we bump the typing_extensions lower bounds here to >=4.7.0?

https://github.com/python/mypy/blob/9a4a5aa77a303c2e56e9dc9bd9327974bf777894/setup.py#L225

https://github.com/python/mypy/blob/9a4a5aa77a303c2e56e9dc9bd9327974bf777894/mypy-requirements.txt#L2

https://github.com/python/mypy/blob/9a4a5aa77a303c2e56e9dc9bd9327974bf777894/pyproject.toml#L9

Reply:
Possibly, but I think it also breaks usage of mypyc-compiled extensions that have typing-extensions 4.7 lying around, which I don't think is something we can guard against.

Reply:
I see. I think it still might be worth bumping the lower bound, though — it will affect freshly compiled extensions. People got unhappy with us when we started using features added in typing_extensions 4.1 without bumping our lower bound: #15487.

(I do agree with you that dropping support for 3.7 is probably the best solution here FWIW.)

Reply:
I guess my preference is: merge this PR now to get CI green (assuming it actually passes). If we still support 3.7 on the next release, bump the typing-extensions lower bounds, otherwise don't.

Reply:
That makes sense, but we need to make sure we remember to do that!